pyanaconda/payload/manager.py

```diff
@@ -60,10 +59,7 @@ class PayloadManager(object):
 
     All states except ERROR are expected to happen linearly, and adding
     a listener for a state that has already been reached or passed will
-    immediately trigger that listener. For example, if the payload thread is
-    currently in DOWNLOADING_GROUP_METADATA, adding a listener for
-    WAITING_NETWORK will immediately run the code being added
-    for WAITING_NETWORK.
+    immediately trigger that listener.
```

Comment:
All these special states were removed, because we don't really need them. The new implementation uses signals for monitoring the stages of the task and progress reporting for showing progress messages about the running task. That's all.